planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -1665,4 +1732,25 @@ void StartPlannerModule::logPullOutStatus(rclcpp::Logger::Level log_level) const
 
   logFunc("=======================================");
 }
+
+StartPlannerModule::StartPlannerData StartPlannerModule::StartPlannerData::clone() const
```

Comment:
I want to keep this method for now. As I'm doing in the body "begin/end of critical section", I'm creating a copy of StartPlannerData. Right now I'm making the copied of each member of StartPlannerData because most of the functions takes `const shared_ptr<PlannerData>` as the argument, not `const PlannerData&`, which is the reason why I'm making the copy of `StartPlannerData.planner_data` as the shared pointer.

In the future refactoring maybe I will replace most of the `const shared_ptr<PlannerData>` arguments to `const PlannerData&`, in which case the `clone()` method is useful.